src/audio/module_adapter/module/cadence.c

```diff
@@ -35,6 +35,24 @@ enum cadence_api_id {
 
 #define DEFAULT_CODEC_ID CADENCE_CODEC_WRAPPER_ID
 
+/* we divide Cadence param id space as follows:
+ * - 0x00000000 - 0x7FFFFFFF, Cadence library ids which come
+ *   from xa_*_dec_api.h
+ * - 0x80000000 - 0xFFFFFFFF, SOF remapping of Cadence library ids
+ */
```

Comment:
Unable to follow the commit message

"
We will do this in two steps:

1) Allow parameters to be specified in topology.
2) Send parameters via snd_codec structure.

This patch addresses step 1).

We need both steps, because some parameters like `pcm width size` for
compress streams is not stored in snd_codec and cannot be retrieved
from the actual compress file.
"

Not following at all which parameters need to be stored in the topology?

Reply:
We need to store in topology parameters which basically cannot be passed via Compress API kernel-userspace interface.

E.g parameters that cannot be stored in `snd_codec` structure.

See: https://github.com/thesofproject/linux/blob/topic/sof-dev/include/uapi/sound/compress_params.h#L410

One such parameter is `PCM width size`, size of PCM output samples in bits. 

The commit message basically says: 

1) add required parameters via topology

2) enhance current implementation to dynamically setup the rest of parameters via `set_param` calback of Compress API.

For example, for AAC we could support various formats (ADTS, ADIF) etc but we currently are limited to only supporting one format (ADTS) as this is the one we could add in the topology.

In the future, I will remove this info from topology and dynamically set it by actually parsing AAC file and send it to FW.

Reply:
not following this either. how is this parameter passed to the firmware? with the set_params?

Reply:
@plbossart yes, via 

```
-> ioctl(compress->fd, SNDRV_COMPRESS_SET_PARAMS, &params)
       -> `sof_compr_set_params`
       
```